--- ICapabilityProviderImpl.java
+++ ICapabilityProviderImpl.java
@@ -12,7 +12,16 @@
 public interface ICapabilityProviderImpl<B extends ICapabilityProviderImpl<B>> extends ICapabilityProvider
 {
-    boolean areCapsCompatible(CapabilityProvider<B> other);
-    boolean areCapsCompatible(@Nullable CapabilityDispatcher other);
-    void invalidateCaps();
-    void reviveCaps();
+    default boolean areCapsCompatible(CapabilityProvider<B> other) {
+        throw new IllegalStateException("should be overridden by mixin");
+    }
+    default boolean areCapsCompatible(@Nullable CapabilityDispatcher other) {
+        throw new IllegalStateException("should be overridden by mixin");
+    }
+    default void invalidateCaps() {
+        // we don't throw anything here because otherwise
+        // Porting Lib will go into my house and murder me
+    }
+    default void reviveCaps() {
+        throw new IllegalStateException("should be overridden by mixin");
+    }
 }
