--- FluidHandlerBlockEntity.java
+++ FluidHandlerBlockEntity.java
@@ -13,4 +13,6 @@
 import net.minecraft.core.Direction;
 import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityDispatcher;
+import net.minecraftforge.common.capabilities.CapabilityProvider;
 import net.minecraftforge.common.capabilities.ForgeCapabilities;
 import net.minecraftforge.common.util.LazyOptional;
@@ -51,5 +53,10 @@
         if (capability == ForgeCapabilities.FLUID_HANDLER)
             return holder.cast();
-        return super.getCapability(capability, facing);
+        return this.getCapability(capability, facing);
     }
+
+    @Override
+    public CompoundTag getPersistentData() {
+        return null;
+    }
 }
