--- LivingEvent.java
+++ LivingEvent.java
@@ -24,4 +24,9 @@
 public class LivingEvent extends EntityEvent
 {
+    public LivingEvent() {
+        super(null);
+        livingEntity = null;
+    }
+
     private final LivingEntity livingEntity;
 
@@ -53,4 +58,5 @@
     public static class LivingTickEvent extends LivingEvent
     {
+        public LivingTickEvent() {}
         public LivingTickEvent(LivingEntity e){ super(e); }
     }
@@ -72,4 +78,5 @@
     public static class LivingJumpEvent extends LivingEvent
     {
+        public LivingJumpEvent() {}
         public LivingJumpEvent(LivingEntity e){ super(e); }
     }
@@ -81,4 +88,7 @@
         private final Entity lookingEntity;
 
+        public LivingVisibilityEvent() {
+            lookingEntity = null;
+        }
         public LivingVisibilityEvent(LivingEntity livingEntity, @Nullable Entity lookingEntity, double originalMultiplier)
         {
