--- LivingConversionEvent.java
+++ LivingConversionEvent.java
@@ -14,4 +14,8 @@
 public class LivingConversionEvent extends LivingEvent
 {
+    public LivingConversionEvent() {
+        super(null);
+    }
+
     public LivingConversionEvent(LivingEntity entity)
     {
@@ -36,4 +40,9 @@
         private final Consumer<Integer> timer;
 
+        public Pre() {
+            outcome = null;
+            timer = null;
+        }
+
         public Pre(LivingEntity entity, EntityType<? extends LivingEntity> outcome, Consumer<Integer> timer)
         {
@@ -76,4 +85,8 @@
         private final LivingEntity outcome;
 
+        public Post() {
+            outcome = null;
+        }
+
         public Post(LivingEntity entity, LivingEntity outcome)
         {
