--- MobEffectEvent.java
+++ MobEffectEvent.java
@@ -25,4 +25,8 @@
     protected final MobEffectInstance effectInstance;
 
+    public MobEffectEvent() {
+        effectInstance = null;
+    }
+
     public MobEffectEvent(LivingEntity living, MobEffectInstance effectInstance)
     {
@@ -47,4 +51,8 @@
         private final MobEffect effect;
 
+        public Remove() {
+            effect = null;
+        }
+
         public Remove(LivingEntity living, MobEffect effect)
         {
@@ -90,4 +98,6 @@
     public static class Applicable extends MobEffectEvent
     {
+        public Applicable() {}
+
         public Applicable(LivingEntity living, @NotNull MobEffectInstance effectInstance)
         {
@@ -114,4 +124,9 @@
         private final Entity source;
 
+        public Added() {
+            oldEffectInstance = null;
+            source = null;
+        }
+
         public Added(LivingEntity living, MobEffectInstance oldEffectInstance, MobEffectInstance newEffectInstance, Entity source)
         {
@@ -157,4 +172,5 @@
     public static class Expired extends MobEffectEvent
     {
+        public Expired() {}
         public Expired(LivingEntity living, MobEffectInstance effectInstance)
         {
