--- UnbakedGeometryHelper.java
+++ UnbakedGeometryHelper.java
@@ -98,7 +96,7 @@
     public static BakedModel bake(BlockModel blockModel, ModelBakery modelBakery, BlockModel owner, Function<Material, TextureAtlasSprite> spriteGetter, ModelState modelState, ResourceLocation modelLocation, boolean guiLight3d)
     {
-        IUnbakedGeometry<?> customModel = blockModel.customData.getCustomGeometry();
+        var customModel = blockModel.getGeometry().getCustomGeometry();
         if (customModel != null)
-            return customModel.bake(blockModel.customData, modelBakery, spriteGetter, modelState, blockModel.getOverrides(modelBakery, owner, spriteGetter), modelLocation);
+            return customModel.bake(blockModel.getGeometry(), modelBakery, spriteGetter, modelState, blockModel.getOverrides(modelBakery, owner, spriteGetter), modelLocation);
 
         // Handle vanilla item models here, since vanilla has a shortcut for them
@@ -125,5 +123,5 @@
 
         var elementsModel = new ElementsModel(blockModel.getElements());
-        return elementsModel.bake(blockModel.customData, modelBakery, spriteGetter, modelState, blockModel.getOverrides(modelBakery, owner, spriteGetter), modelLocation);
+        return elementsModel.bake(blockModel.getGeometry(), modelBakery, spriteGetter, modelState, blockModel.getOverrides(modelBakery, owner, spriteGetter), modelLocation);
     }
 
