--- ClientChatReceivedEvent.java
+++ ClientChatReceivedEvent.java
@@ -38,4 +37,10 @@
     private final MessageSigner messageSigner;
 
+    public ClientChatReceivedEvent() {
+        boundChatType = null;
+        playerChatMessage = null;
+        messageSigner = null;
+    }
+
     @ApiStatus.Internal
     public ClientChatReceivedEvent(ChatType.Bound boundChatType, Component message, PlayerChatMessage playerChatMessage, MessageSigner messageSigner)
