--- ClientPlayerChangeGameTypeEvent.java
+++ ClientPlayerChangeGameTypeEvent.java
@@ -28,4 +28,9 @@
     private final GameType newGameType;
 
+    public ClientPlayerChangeGameTypeEvent() {
+        info = null;
+        newGameType = currentGameType = null;
+    }
+
     @ApiStatus.Internal
     public ClientPlayerChangeGameTypeEvent(PlayerInfo info, GameType currentGameType, GameType newGameType)
