--- ClientPlayerNetworkEvent.java
+++ ClientPlayerNetworkEvent.java
@@ -33,4 +33,10 @@
     private final Connection connection;
 
+    public ClientPlayerNetworkEvent() {
+        multiPlayerGameMode = null;
+        player = null;
+        connection = null;
+    }
+
     @ApiStatus.Internal
     protected ClientPlayerNetworkEvent(final MultiPlayerGameMode multiPlayerGameMode, final LocalPlayer player, final Connection connection)
@@ -75,4 +81,6 @@
     public static class LoggingIn extends ClientPlayerNetworkEvent
     {
+        public LoggingIn() {}
+
         @ApiStatus.Internal
         public LoggingIn(final MultiPlayerGameMode controller, final LocalPlayer player, final Connection networkManager)
@@ -94,4 +102,6 @@
     public static class LoggingOut extends ClientPlayerNetworkEvent
     {
+        public LoggingOut() {}
+
         @ApiStatus.Internal
         public LoggingOut(@Nullable final MultiPlayerGameMode controller, @Nullable final LocalPlayer player, @Nullable final Connection networkManager)
@@ -150,4 +160,8 @@
         private final LocalPlayer oldPlayer;
 
+        public Clone() {
+            oldPlayer = null;
+        }
+
         @ApiStatus.Internal
         public Clone(final MultiPlayerGameMode pc, final LocalPlayer oldPlayer, final LocalPlayer newPlayer, final Connection networkManager)
