--- ContainerScreenEvent.java
+++ ContainerScreenEvent.java
@@ -28,4 +28,8 @@
     private final AbstractContainerScreen<?> containerScreen;
 
+    public ContainerScreenEvent() {
+        containerScreen = null;
+    }
+
     @ApiStatus.Internal
     protected ContainerScreenEvent(AbstractContainerScreen<?> containerScreen)
@@ -58,4 +62,9 @@
         private final int mouseY;
 
+        public Render() {
+            poseStack = null;
+            mouseX = mouseY = 0;
+        }
+
         @ApiStatus.Internal
         protected Render(AbstractContainerScreen<?> guiContainer, PoseStack poseStack, int mouseX, int mouseY)
@@ -105,4 +114,6 @@
         public static class Foreground extends Render
         {
+            public Foreground() {}
+
             @ApiStatus.Internal
             public Foreground(AbstractContainerScreen<?> guiContainer, PoseStack poseStack, int mouseX, int mouseY)
@@ -123,4 +134,6 @@
         public static class Background extends Render
         {
+            public Background() {}
+
             @ApiStatus.Internal
             public Background(AbstractContainerScreen<?> guiContainer, PoseStack poseStack, int mouseX, int mouseY)
