--- CustomizeGuiOverlayEvent.java
+++ CustomizeGuiOverlayEvent.java
@@ -30,4 +30,10 @@
     private final float partialTick;
 
+    public CustomizeGuiOverlayEvent() {
+        window = null;
+        poseStack = null;
+        partialTick = 0F;
+    }
+
     @ApiStatus.Internal
     protected CustomizeGuiOverlayEvent(Window window, PoseStack poseStack, float partialTick)
@@ -70,4 +76,9 @@
         private int increment;
 
+        public BossEventProgress() {
+            bossEvent = null;
+            x = y = 0;
+        }
+
         @ApiStatus.Internal
         public BossEventProgress(Window window, PoseStack poseStack, float partialTick, LerpingBossEvent bossEvent, int x, int y, int increment)
@@ -137,4 +148,8 @@
         private final ArrayList<String> right;
 
+        public DebugText() {
+            left = right = null;
+        }
+
         @ApiStatus.Internal
         public DebugText(Window window, PoseStack poseStack, float partialTick, ArrayList<String> left, ArrayList<String> right)
@@ -175,4 +190,6 @@
         private int posY;
 
+        public Chat() {}
+
         @ApiStatus.Internal
         public Chat(Window window, PoseStack poseStack, float partialTick, int posX, int posY)
