--- InputEvent.java
+++ InputEvent.java
@@ -48,4 +48,8 @@
         private final int modifiers;
 
+        public MouseButton() {
+            button = action = modifiers = 0;
+        }
+
         @ApiStatus.Internal
         protected MouseButton(int button, int action, int modifiers)
@@ -108,4 +112,5 @@
         public static class Pre extends MouseButton
         {
+            public Pre() {}
             @ApiStatus.Internal
             public Pre(int button, int action, int modifiers)
@@ -127,4 +132,5 @@
         public static class Post extends MouseButton
         {
+            public Post() {}
             @ApiStatus.Internal
             public Post(int button, int action, int modifiers)
@@ -157,4 +163,9 @@
         private final boolean rightDown;
 
+        public MouseScrollingEvent() {
+            scrollDelta = mouseX = mouseY = 0D;
+            leftDown = middleDown = rightDown = false;
+        }
+
         @ApiStatus.Internal
         public MouseScrollingEvent(double scrollDelta, boolean leftDown, boolean middleDown, boolean rightDown, double mouseX, double mouseY)
@@ -232,4 +243,8 @@
         private final int modifiers;
 
+        public Key() {
+            key = scanCode = action = modifiers = 0;
+        }
+
         @ApiStatus.Internal
         public Key(int key, int scanCode, int action, int modifiers)
@@ -321,4 +336,10 @@
         private boolean handSwing = true;
 
+        public InteractionKeyMappingTriggered() {
+            button = -1;
+            keyMapping = null;
+            hand = null;
+        }
+
         @ApiStatus.Internal
         public InteractionKeyMappingTriggered(int button, KeyMapping keyMapping, InteractionHand hand)
