--- ModelEvent.java
+++ ModelEvent.java
@@ -48,4 +48,10 @@
         private final ModelBakery modelBakery;
 
+        public BakingCompleted() {
+            modelManager = null;
+            models = null;
+            modelBakery = null;
+        }
+
         @ApiStatus.Internal
         public BakingCompleted(ModelManager modelManager, Map<ResourceLocation, BakedModel> models, ModelBakery modelBakery)
@@ -94,4 +100,8 @@
         private final Set<ResourceLocation> models;
 
+        public RegisterAdditional() {
+            models = null;
+        }
+
         @ApiStatus.Internal
         public RegisterAdditional(Set<ResourceLocation> models)
@@ -119,8 +129,12 @@
     public static class RegisterGeometryLoaders extends ModelEvent implements IModBusEvent
     {
-        private final Map<ResourceLocation, IGeometryLoader<?>> loaders;
+        private final Map<ResourceLocation, io.github.fabricators_of_create.porting_lib.model.geometry.IGeometryLoader<?>> loaders;
 
+        public RegisterGeometryLoaders() {
+            loaders = null;
+        }
+
         @ApiStatus.Internal
-        public RegisterGeometryLoaders(Map<ResourceLocation, IGeometryLoader<?>> loaders)
+        public RegisterGeometryLoaders(Map<ResourceLocation, io.github.fabricators_of_create.porting_lib.model.geometry.IGeometryLoader<?>> loaders)
         {
             this.loaders = loaders;
