--- RegisterKeyMappingsEvent.java
+++ RegisterKeyMappingsEvent.java
@@ -6,4 +6,5 @@
 package net.minecraftforge.client.event;
 
+import net.fabricmc.fabric.api.client.keybinding.v1.KeyBindingHelper;
 import net.minecraft.client.KeyMapping;
 import net.minecraft.client.Options;
@@ -39,5 +40,5 @@
     public void register(KeyMapping key)
     {
-        options.keyMappings = ArrayUtils.add(options.keyMappings, key);
+        KeyBindingHelper.registerKeyBinding(key);
     }
 }
