--- RegisterClientTooltipComponentFactoriesEvent.java
+++ RegisterClientTooltipComponentFactoriesEvent.java
@@ -31,4 +31,8 @@
     private final Map<Class<? extends TooltipComponent>, Function<TooltipComponent, ClientTooltipComponent>> factories;
 
+    public RegisterClientTooltipComponentFactoriesEvent() {
+        factories = null;
+    }
+
     @ApiStatus.Internal
     public RegisterClientTooltipComponentFactoriesEvent(Map<Class<? extends TooltipComponent>, Function<TooltipComponent, ClientTooltipComponent>> factories)
