--- RegisterRecipeBookCategoriesEvent.java
+++ RegisterRecipeBookCategoriesEvent.java
@@ -36,4 +36,10 @@
     private final Map<RecipeType<?>, Function<Recipe<?>, RecipeBookCategories>> recipeCategoryLookups;
 
+    public RegisterRecipeBookCategoriesEvent() {
+        aggregateCategories = null;
+        typeCategories = null;
+        recipeCategoryLookups = null;
+    }
+
     @ApiStatus.Internal
     public RegisterRecipeBookCategoriesEvent(
