--- RegisterGuiOverlaysEvent.java
+++ RegisterGuiOverlaysEvent.java
@@ -36,4 +36,9 @@
     private final List<ResourceLocation> orderedOverlays;
 
+    public RegisterGuiOverlaysEvent() {
+        overlays = null;
+        orderedOverlays = null;
+    }
+
     @ApiStatus.Internal
     public RegisterGuiOverlaysEvent(Map<ResourceLocation, IGuiOverlay> overlays, List<ResourceLocation> orderedOverlays)
