--- RenderHandEvent.java
+++ RenderHandEvent.java
@@ -41,4 +41,13 @@
     private final ItemStack stack;
 
+    public RenderHandEvent() {
+        hand = null;
+        multiBufferSource = null;
+        poseStack = null;
+        packedLight = -1;
+        partialTick = interpolatedPitch = swingProgress = equipProgress = 0F;
+        stack = null;
+    }
+
     @ApiStatus.Internal
     public RenderHandEvent(InteractionHand hand, PoseStack poseStack, MultiBufferSource multiBufferSource, int packedLight,
