--- RenderGuiOverlayEvent.java
+++ RenderGuiOverlayEvent.java
@@ -31,4 +31,11 @@
     private final NamedGuiOverlay overlay;
 
+    public RenderGuiOverlayEvent() {
+        window = null;
+        poseStack = null;
+        partialTick = 0F;
+        overlay = null;
+    }
+
     @ApiStatus.Internal
     protected RenderGuiOverlayEvent(Window window, PoseStack poseStack, float partialTick, NamedGuiOverlay overlay)
@@ -75,4 +82,6 @@
     public static class Pre extends RenderGuiOverlayEvent
     {
+        public Pre() {}
+
         @ApiStatus.Internal
         public Pre(Window window, PoseStack poseStack, float partialTick, NamedGuiOverlay overlay)
@@ -92,4 +101,6 @@
     public static class Post extends RenderGuiOverlayEvent
     {
+        public Post() {}
+
         @ApiStatus.Internal
         public Post(Window window, PoseStack poseStack, float partialTick, NamedGuiOverlay overlay)
