--- RenderLivingEvent.java
+++ RenderLivingEvent.java
@@ -41,4 +41,13 @@
     private final int packedLight;
 
+    public RenderLivingEvent() {
+        entity = null;
+        renderer = null;
+        partialTick = 0F;
+        poseStack = null;
+        multiBufferSource = null;
+        packedLight = 0;
+    }
+
     @ApiStatus.Internal
     protected RenderLivingEvent(LivingEntity entity, LivingEntityRenderer<T, M> renderer, float partialTick, PoseStack poseStack,
@@ -120,4 +129,6 @@
     public static class Pre<T extends LivingEntity, M extends EntityModel<T>> extends RenderLivingEvent<T, M>
     {
+        public Pre() {}
+
         @ApiStatus.Internal
         public Pre(LivingEntity entity, LivingEntityRenderer<T, M> renderer, float partialTick, PoseStack poseStack, MultiBufferSource multiBufferSource, int packedLight)
@@ -140,4 +151,6 @@
     public static class Post<T extends LivingEntity, M extends EntityModel<T>> extends RenderLivingEvent<T, M>
     {
+        public Post() {}
+
         @ApiStatus.Internal
         public Post(LivingEntity entity, LivingEntityRenderer<T, M> renderer, float partialTick, PoseStack poseStack, MultiBufferSource multiBufferSource, int packedLight)
