--- RenderLevelStageEvent.java
+++ RenderLevelStageEvent.java
@@ -20,6 +20,10 @@
 import net.minecraft.resources.ResourceLocation;
 import net.minecraftforge.client.ForgeRenderTypes;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Cancelable;
 import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.LogicalSide;
 import net.minecraftforge.fml.event.IModBusEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
 
 /**
@@ -42,5 +46,16 @@
     private final Camera camera;
     private final Frustum frustum;
+    public RenderLevelStageEvent() {
+        stage = null;
 
+        levelRenderer = null;
+        poseStack = null;
+        projectionMatrix = null;
+        renderTick = 0;
+        partialTick = 0F;
+        camera = null;
+        frustum = null;
+    }
+
     public RenderLevelStageEvent(Stage stage, LevelRenderer levelRenderer, PoseStack poseStack, Matrix4f projectionMatrix, int renderTick, float partialTick, Camera camera, Frustum frustum)
     {
