--- RenderBlockScreenEffectEvent.java
+++ RenderBlockScreenEffectEvent.java
@@ -55,4 +55,12 @@
     private final BlockPos blockPos;
 
+    public RenderBlockScreenEffectEvent() {
+        player = null;
+        poseStack = null;
+        overlayType = null;
+        blockPos = null;
+        blockState = null;
+    }
+
     @ApiStatus.Internal
     public RenderBlockScreenEffectEvent(Player player, PoseStack poseStack, OverlayType type, BlockState block, BlockPos blockPos)
