--- RenderTooltipEvent.java
+++ RenderTooltipEvent.java
@@ -42,4 +42,10 @@
     protected final List<ClientTooltipComponent> components;
 
+    public RenderTooltipEvent() {
+        itemStack = ItemStack.EMPTY;
+        poseStack = null;
+        components = null;
+    }
+
     @ApiStatus.Internal
     protected RenderTooltipEvent(@NotNull ItemStack itemStack, PoseStack poseStack, int x, int y, @NotNull Font font, @NotNull List<ClientTooltipComponent> components)
@@ -127,4 +133,10 @@
         private int maxWidth;
 
+        public GatherComponents() {
+            itemStack = null;
+            screenHeight = screenWidth = 0;
+            tooltipElements = null;
+        }
+
         @ApiStatus.Internal
         public GatherComponents(ItemStack itemStack, int screenWidth, int screenHeight, List<Either<FormattedText, TooltipComponent>> tooltipElements, int maxWidth)
@@ -213,4 +225,8 @@
         private final int screenHeight;
 
+        public Pre() {
+            screenHeight = screenWidth = 0;
+        }
+
         @ApiStatus.Internal
         public Pre(@NotNull ItemStack stack, PoseStack poseStack, int x, int y, int screenWidth, int screenHeight, @NotNull Font font, @NotNull List<ClientTooltipComponent> components)
@@ -290,4 +306,8 @@
         private int borderEnd;
 
+        public Color() {
+            originalBackground = originalBorderStart = originalBorderEnd = 0;
+        }
+
         @ApiStatus.Internal
         public Color(@NotNull ItemStack stack, PoseStack poseStack, int x, int y, @NotNull Font fr, int background, int borderStart, int borderEnd, @NotNull List<ClientTooltipComponent> components)
