--- RenderItemInFrameEvent.java
+++ RenderItemInFrameEvent.java
@@ -40,4 +40,13 @@
     private final int packedLight;
 
+    public RenderItemInFrameEvent() {
+        itemStack = null;
+        itemFrameEntity = null;
+        renderer = null;
+        poseStack = null;
+        multiBufferSource = null;
+        packedLight = 0;
+    }
+
     @ApiStatus.Internal
     public RenderItemInFrameEvent(ItemFrame itemFrame, ItemFrameRenderer<?> renderItemFrame, PoseStack poseStack,
