--- ScreenEvent.java
+++ ScreenEvent.java
@@ -44,4 +44,8 @@
     private final Screen screen;
 
+    public ScreenEvent() {
+        screen = null;
+    }
+
     @ApiStatus.Internal
     protected ScreenEvent(Screen screen)
@@ -76,4 +80,9 @@
         private final List<GuiEventListener> listenerList;
 
+        public Init() {
+            add = remove = null;
+            listenerList = null;
+        }
+
         @ApiStatus.Internal
         protected Init(Screen screen, List<GuiEventListener> listenerList, Consumer<GuiEventListener> add, Consumer<GuiEventListener> remove)
@@ -126,4 +135,6 @@
         public static class Pre extends Init
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, List<GuiEventListener> list, Consumer<GuiEventListener> add, Consumer<GuiEventListener> remove)
@@ -143,4 +154,6 @@
         public static class Post extends Init
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, List<GuiEventListener> list, Consumer<GuiEventListener> add, Consumer<GuiEventListener> remove)
@@ -165,4 +178,10 @@
         private final float partialTick;
 
+        public Render() {
+            poseStack = null;
+            mouseX = mouseY = 0;
+            partialTick = 0F;
+        }
+
         @ApiStatus.Internal
         protected Render(Screen screen, PoseStack poseStack, int mouseX, int mouseY, float partialTick)
@@ -219,4 +238,6 @@
         public static class Pre extends Render
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, PoseStack poseStack, int mouseX, int mouseY, float partialTick)
@@ -236,4 +257,6 @@
         public static class Post extends Render
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, PoseStack poseStack, int mouseX, int mouseY, float partialTick)
@@ -257,4 +280,8 @@
         private final PoseStack poseStack;
 
+        public BackgroundRendered() {
+            poseStack = null;
+        }
+
         @ApiStatus.Internal
         public BackgroundRendered(Screen screen, PoseStack poseStack)
@@ -291,4 +318,8 @@
         private int horizontalOffset;
 
+        public RenderInventoryMobEffects() {
+            availableSpace = -1;
+        }
+
         @ApiStatus.Internal
         public RenderInventoryMobEffects(Screen screen, int availableSpace, boolean compact, int horizontalOffset)
@@ -363,4 +394,8 @@
         private final double mouseY;
 
+        public MouseInput() {
+            mouseX = mouseY = 0D;
+        }
+
         @ApiStatus.Internal
         protected MouseInput(Screen screen, double mouseX, double mouseY)
@@ -399,4 +434,8 @@
         private final int button;
 
+        public MouseButtonPressed() {
+            button = -1;
+        }
+
         @ApiStatus.Internal
         public MouseButtonPressed(Screen screen, double mouseX, double mouseY, int button)
@@ -430,4 +469,6 @@
         public static class Pre extends MouseButtonPressed
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, double mouseX, double mouseY, int button)
@@ -457,4 +498,8 @@
             private final boolean handled;
 
+            public Post() {
+                handled = false;
+            }
+
             @ApiStatus.Internal
             public Post(Screen screen, double mouseX, double mouseY, int button, boolean handled)
@@ -485,4 +530,8 @@
         private final int button;
 
+        public MouseButtonReleased() {
+            button = -1;
+        }
+
         @ApiStatus.Internal
         public MouseButtonReleased(Screen screen, double mouseX, double mouseY, int button)
@@ -516,4 +565,6 @@
         public static class Pre extends MouseButtonReleased
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, double mouseX, double mouseY, int button)
@@ -543,4 +594,8 @@
             private final boolean handled;
 
+            public Post() {
+                handled = false;
+            }
+
             @ApiStatus.Internal
             public Post(Screen screen, double mouseX, double mouseY, int button, boolean handled)
@@ -573,4 +628,9 @@
         private final double dragY;
 
+        public MouseDragged() {
+            mouseButton = -1;
+            dragX = dragY = 0D;
+        }
+
         @ApiStatus.Internal
         public MouseDragged(Screen screen, double mouseX, double mouseY, int mouseButton, double dragX, double dragY)
@@ -622,4 +682,6 @@
         public static class Pre extends MouseDragged
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, double mouseX, double mouseY, int mouseButton, double dragX, double dragY)
@@ -641,4 +703,6 @@
         public static class Post extends MouseDragged
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, double mouseX, double mouseY, int mouseButton, double dragX, double dragY)
@@ -660,4 +724,8 @@
         private final double scrollDelta;
 
+        public MouseScrolled() {
+            scrollDelta = 0D;
+        }
+
         @ApiStatus.Internal
         public MouseScrolled(Screen screen, double mouseX, double mouseY, double scrollDelta)
@@ -688,4 +756,6 @@
         public static class Pre extends MouseScrolled
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, double mouseX, double mouseY, double scrollDelta)
@@ -707,4 +777,6 @@
         public static class Post extends MouseScrolled
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, double mouseX, double mouseY, double scrollDelta)
@@ -730,4 +802,8 @@
         private final int modifiers;
 
+        public KeyInput() {
+            keyCode = scanCode = modifiers = 0;
+        }
+
         @ApiStatus.Internal
         protected KeyInput(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -791,4 +867,6 @@
     public static abstract class KeyPressed extends KeyInput
     {
+        public KeyPressed() {}
+
         @ApiStatus.Internal
         public KeyPressed(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -810,4 +888,6 @@
         public static class Pre extends KeyPressed
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -830,4 +910,6 @@
         public static class Post extends KeyPressed
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -847,4 +929,6 @@
     public static abstract class KeyReleased extends KeyInput
     {
+        public KeyReleased() {}
+
         @ApiStatus.Internal
         public KeyReleased(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -866,4 +950,6 @@
         public static class Pre extends KeyReleased
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -886,4 +972,6 @@
         public static class Post extends KeyReleased
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, int keyCode, int scanCode, int modifiers)
@@ -907,4 +995,9 @@
         private final int modifiers;
 
+        public CharacterTyped() {
+            codePoint = 0;
+            modifiers = 0;
+        }
+
         @ApiStatus.Internal
         public CharacterTyped(Screen screen, char codePoint, int modifiers)
@@ -952,4 +1045,6 @@
         public static class Pre extends CharacterTyped
         {
+            public Pre() {}
+
             @ApiStatus.Internal
             public Pre(Screen screen, char codePoint, int modifiers)
@@ -971,4 +1066,6 @@
         public static class Post extends CharacterTyped
         {
+            public Post() {}
+
             @ApiStatus.Internal
             public Post(Screen screen, char codePoint, int modifiers)
@@ -998,4 +1095,8 @@
         private Screen newScreen;
 
+        public Opening() {
+            currentScreen = null;
+        }
+
         @ApiStatus.Internal
         public Opening(@Nullable Screen currentScreen, Screen screen)
@@ -1046,4 +1147,6 @@
     public static class Closing extends ScreenEvent
     {
+        public Closing() {}
+
         @ApiStatus.Internal
         public Closing(Screen screen)
