--- TextureAtlasSpriteLoaderManager.java
+++ TextureAtlasSpriteLoaderManager.java
@@ -39,5 +39,5 @@
         var loaders = new HashMap<ResourceLocation, ITextureAtlasSpriteLoader>();
         var event = new RegisterTextureAtlasSpriteLoadersEvent(loaders);
-        ModLoader.get().postEventWithWrapInModOrder(event, (mc, e) -> ModLoadingContext.get().setActiveContainer(mc), (mc, e) -> ModLoadingContext.get().setActiveContainer(null));
+        ModLoader.get().kiltPostEventWrappingMods(event);
         LOADERS = ImmutableMap.copyOf(loaders);
     }
