--- BlockEvent.java
+++ BlockEvent.java
@@ -26,4 +26,5 @@
 import net.minecraftforge.common.ToolAction;
 import net.minecraftforge.common.ToolActions;
+import net.minecraftforge.common.extensions.IForgeBlockState;
 import net.minecraftforge.common.util.BlockSnapshot;
 import net.minecraftforge.eventbus.api.Cancelable;
@@ -87,5 +88,5 @@
                 int fortuneLevel = EnchantmentHelper.getItemEnchantmentLevel(Enchantments.BLOCK_FORTUNE, player.getMainHandItem());
                 int silkTouchLevel = EnchantmentHelper.getItemEnchantmentLevel(Enchantments.SILK_TOUCH, player.getMainHandItem());
-                this.exp = state.getExpDrop(level, level.random, pos, fortuneLevel, silkTouchLevel);
+                this.exp = ((IForgeBlockState) state).getExpDrop(level, level.random, pos, fortuneLevel, silkTouchLevel);
             }
         }
