--- BlockEvent.java
+++ BlockEvent.java
@@ -26,4 +26,5 @@
 import net.minecraftforge.common.ToolAction;
 import net.minecraftforge.common.ToolActions;
+import net.minecraftforge.common.extensions.IForgeBlockState;
 import net.minecraftforge.common.util.BlockSnapshot;
 import net.minecraftforge.eventbus.api.Cancelable;
@@ -41,4 +42,11 @@
     private final BlockPos pos;
     private final BlockState state;
+
+    public BlockEvent() {
+        level = null;
+        pos = null;
+        state = null;
+    }
+
     public BlockEvent(LevelAccessor level, BlockPos pos, BlockState state)
     {
@@ -74,4 +82,8 @@
         private int exp;
 
+        public BreakEvent() {
+            player = null;
+        }
+
         public BreakEvent(Level level, BlockPos pos, BlockState state, Player player)
         {
@@ -87,5 +99,5 @@
                 int fortuneLevel = EnchantmentHelper.getItemEnchantmentLevel(Enchantments.BLOCK_FORTUNE, player.getMainHandItem());
                 int silkTouchLevel = EnchantmentHelper.getItemEnchantmentLevel(Enchantments.SILK_TOUCH, player.getMainHandItem());
-                this.exp = state.getExpDrop(level, level.random, pos, fortuneLevel, silkTouchLevel);
+                this.exp = ((IForgeBlockState) state).getExpDrop(level, level.random, pos, fortuneLevel, silkTouchLevel);
             }
         }
@@ -130,4 +142,10 @@
         private final BlockState placedAgainst;
 
+        public EntityPlaceEvent() {
+            entity = null;
+            blockSnapshot = null;
+            placedAgainst = placedBlock = null;
+        }
+
         public EntityPlaceEvent(@NotNull BlockSnapshot blockSnapshot, @NotNull BlockState placedAgainst, @Nullable Entity entity)
         {
@@ -163,4 +181,8 @@
         private final List<BlockSnapshot> blockSnapshots;
 
+        public EntityMultiPlaceEvent() {
+            blockSnapshots = null;
+        }
+
         public EntityMultiPlaceEvent(@NotNull List<BlockSnapshot> blockSnapshots, @NotNull BlockState placedAgainst, @Nullable Entity entity) {
             super(blockSnapshots.get(0), placedAgainst, entity);
@@ -195,4 +217,9 @@
         private final boolean forceRedstoneUpdate;
 
+        public NeighborNotifyEvent() {
+            notifiedSides = null;
+            forceRedstoneUpdate = false;
+        }
+
         public NeighborNotifyEvent(Level level, BlockPos pos, BlockState state, EnumSet<Direction> notifiedSides, boolean forceRedstoneUpdate)
         {
@@ -235,4 +262,10 @@
         private final BlockState state;
 
+        public CreateFluidSourceEvent() {
+            level = null;
+            pos = null;
+            state = null;
+        }
+
         public CreateFluidSourceEvent(LevelReader level, BlockPos pos, BlockState state)
         {
@@ -273,4 +306,8 @@
         private BlockState origState;
 
+        public FluidPlaceBlockEvent() {
+            liquidPos = null;
+        }
+
         public FluidPlaceBlockEvent(LevelAccessor level, BlockPos pos, BlockPos liquidPos, BlockState state)
         {
@@ -317,4 +354,5 @@
     public static class CropGrowEvent extends BlockEvent
     {
+        public CropGrowEvent() {}
         public CropGrowEvent(Level level, BlockPos pos, BlockState state)
         {
@@ -336,4 +374,5 @@
         public static class Pre extends CropGrowEvent
         {
+            public Pre() {}
             public Pre(Level level, BlockPos pos, BlockState state)
             {
@@ -354,4 +393,8 @@
         {
             private final BlockState originalState;
+            public Post() {
+                originalState = null;
+            }
+
             public Post(Level level, BlockPos pos, BlockState original, BlockState state)
             {
@@ -378,4 +421,9 @@
         private final float fallDistance;
 
+        public FarmlandTrampleEvent() {
+            entity = null;
+            fallDistance = 0F;
+        }
+
         public FarmlandTrampleEvent(Level level, BlockPos pos, BlockState state, float fallDistance, Entity entity)
         {
@@ -405,4 +453,8 @@
         private final PortalShape size;
 
+        public PortalSpawnEvent() {
+            size = null;
+        }
+
         public PortalSpawnEvent(LevelAccessor level, BlockPos pos, BlockState state, PortalShape size)
         {
@@ -435,4 +487,10 @@
         private BlockState state;
 
+        public BlockToolModificationEvent() {
+            context = null;
+            toolAction = null;
+            simulate = false;
+        }
+
         public BlockToolModificationEvent(BlockState originalState, @NotNull UseOnContext context, ToolAction toolAction, boolean simulate)
         {
