--- ChunkDataEvent.java
+++ ChunkDataEvent.java
@@ -31,4 +31,8 @@
     private final CompoundTag data;
 
+    public ChunkDataEvent() {
+        data = null;
+    }
+
     public ChunkDataEvent(ChunkAccess chunk, CompoundTag data)
     {
@@ -63,4 +67,6 @@
         private ChunkStatus.ChunkType status;
 
+        public Load() {}
+
         public Load(ChunkAccess chunk, CompoundTag data, ChunkStatus.ChunkType status)
         {
@@ -88,4 +94,5 @@
     public static class Save extends ChunkDataEvent
     {
+        public Save() {}
         public Save(ChunkAccess chunk, LevelAccessor world, CompoundTag data)
         {
