--- ChunkWatchEvent.java
+++ ChunkWatchEvent.java
@@ -32,4 +32,10 @@
     private final ChunkPos pos;
 
+    public ChunkWatchEvent() {
+        level = null;
+        player = null;
+        pos = null;
+    }
+
     public ChunkWatchEvent(ServerPlayer player, ChunkPos pos, ServerLevel level)
     {
@@ -81,4 +87,8 @@
         private final LevelChunk chunk;
 
+        public Watch() {
+            chunk = null;
+        }
+
         public Watch(ServerPlayer player, LevelChunk chunk, ServerLevel level)
         {
@@ -106,4 +116,6 @@
     public static class UnWatch extends ChunkWatchEvent
     {
+        public UnWatch() {}
+
         public UnWatch(ServerPlayer player, ChunkPos pos, ServerLevel level) {super(player, pos, level);}
     }
