--- ChunkTicketLevelUpdatedEvent.java
+++ ChunkTicketLevelUpdatedEvent.java
@@ -39,4 +39,11 @@
     private final ChunkHolder chunkHolder;
 
+    public ChunkTicketLevelUpdatedEvent() {
+        level = null;
+        chunkPos = 0L;
+        oldTicketLevel = newTicketLevel = 0;
+        chunkHolder = null;
+    }
+
     public ChunkTicketLevelUpdatedEvent(ServerLevel level, long chunkPos, int oldTicketLevel, int newTicketLevel, @Nullable ChunkHolder chunkHolder)
     {
