--- ChunkEvent.java
+++ ChunkEvent.java
@@ -27,4 +27,8 @@
     private final ChunkAccess chunk;
 
+    public ChunkEvent() {
+        chunk = null;
+    }
+
     public ChunkEvent(ChunkAccess chunk)
     {
@@ -59,4 +63,5 @@
     public static class Load extends ChunkEvent
     {
+        public Load() {}
         public Load(ChunkAccess chunk)
         {
@@ -78,4 +83,5 @@
     public static class Unload extends ChunkEvent
     {
+        public Unload() {}
         public Unload(ChunkAccess chunk)
         {
