--- ExplosionEvent.java
+++ ExplosionEvent.java
@@ -31,4 +31,9 @@
     private final Explosion explosion;
 
+    public ExplosionEvent() {
+        level = null;
+        explosion = null;
+    }
+
     public ExplosionEvent(Level level, Explosion explosion)
     {
@@ -56,4 +61,5 @@
     public static class Start extends ExplosionEvent
     {
+        public Start() {}
         public Start(Level level, Explosion explosion)
         {
@@ -72,4 +78,8 @@
         private final List<Entity> entityList;
 
+        public Detonate() {
+            entityList = null;
+        }
+
         public Detonate(Level level, Explosion explosion, List<Entity> entityList)
         {
