--- NoteBlockEvent.java
+++ NoteBlockEvent.java
@@ -22,4 +22,6 @@
     private int noteId;
 
+    public NoteBlockEvent() {}
+
     protected NoteBlockEvent(Level world, BlockPos pos, BlockState state, int note)
     {
@@ -76,4 +78,6 @@
         private NoteBlockInstrument instrument;
 
+        public Play() {}
+
         public Play(Level world, BlockPos pos, BlockState state, int note, NoteBlockInstrument instrument)
         {
@@ -103,4 +107,9 @@
         private final Octave oldOctave;
 
+        public Change() {
+            oldNote = null;
+            oldOctave = null;
+        }
+
         public Change(Level world, BlockPos pos, BlockState state, int oldNote, int newNote)
         {
