--- SeparateTransformsModel.java
+++ SeparateTransformsModel.java
@@ -208,7 +208,7 @@
             for (ItemTransforms.TransformType transform : ItemTransforms.TransformType.values())
             {
-                if (perspectiveData.has(transform.getSerializeName()))
+                if (perspectiveData.has(transform.name()))
                 {
-                    BlockModel perspectiveModel = deserializationContext.deserialize(GsonHelper.getAsJsonObject(perspectiveData, transform.getSerializeName()), BlockModel.class);
+                    BlockModel perspectiveModel = deserializationContext.deserialize(GsonHelper.getAsJsonObject(perspectiveData, transform.name()), BlockModel.class);
                     perspectives.put(transform, perspectiveModel);
                 }
