--- AdvancementEvent.java
+++ AdvancementEvent.java
@@ -26,4 +26,8 @@
     private final Advancement advancement;
 
+    public AdvancementEvent() {
+        advancement = null;
+    }
+
     public AdvancementEvent(Player player, Advancement advancement)
     {
@@ -55,4 +59,8 @@
         private final Advancement advancement;
 
+        public AdvancementEarnEvent() {
+            advancement = null;
+        }
+
         public AdvancementEarnEvent(Player player, Advancement earned)
         {
@@ -91,4 +99,11 @@
         private final AdvancementEvent.AdvancementProgressEvent.ProgressType progressType;
 
+        public AdvancementProgressEvent() {
+            advancement = null;
+            advancementProgress = null;
+            criterionName = null;
+            progressType = null;
+        }
+
         public AdvancementProgressEvent(Player player, Advancement progressed, AdvancementProgress advancementProgress, String criterionName, AdvancementEvent.AdvancementProgressEvent.ProgressType progressType)
         {
