--- AnvilRepairEvent.java
+++ AnvilRepairEvent.java
@@ -27,4 +27,8 @@
     private float breakChance; // Anvil's chance to break (reduced by 1 durability) when this is complete. Default is 12% (0.12f)
 
+    public AnvilRepairEvent() {
+        left = right = output = ItemStack.EMPTY;
+    }
+
     public AnvilRepairEvent(Player player, @NotNull ItemStack left, @NotNull ItemStack right, @NotNull ItemStack output)
     {
