--- PlayerContainerEvent.java
+++ PlayerContainerEvent.java
@@ -12,4 +12,9 @@
 {
     private final AbstractContainerMenu container;
+
+    public PlayerContainerEvent() {
+        container = null;
+    }
+
     public PlayerContainerEvent(Player player, AbstractContainerMenu container)
     {
@@ -20,4 +25,5 @@
     public static class Open extends PlayerContainerEvent
     {
+        public Open() {}
         public Open(Player player, AbstractContainerMenu container)
         {
@@ -27,4 +33,5 @@
     public static class Close extends PlayerContainerEvent
     {
+        public Close() {}
         public Close(Player player, AbstractContainerMenu container)
         {
