--- PlayerInteractEvent.java
+++ PlayerInteractEvent.java
@@ -46,4 +46,10 @@
     private InteractionResult cancellationResult = InteractionResult.PASS;
 
+    public PlayerInteractEvent() {
+        hand = null;
+        pos = null;
+        face = null;
+    }
+
     private PlayerInteractEvent(Player player, InteractionHand hand, BlockPos pos, @Nullable Direction face)
     {
@@ -69,4 +75,9 @@
         private final Entity target;
 
+        public EntityInteractSpecific() {
+            localPos = null;
+            target = null;
+        }
+
         public EntityInteractSpecific(Player player, InteractionHand hand, Entity target, Vec3 localPos)
         {
@@ -111,4 +122,8 @@
         private final Entity target;
 
+        public EntityInteract() {
+            target = null;
+        }
+
         public EntityInteract(Player player, InteractionHand hand, Entity target)
         {
@@ -143,4 +158,7 @@
         private BlockHitResult hitVec;
 
+        public RightClickBlock() {
+        }
+
         public RightClickBlock(Player player, InteractionHand hand, BlockPos pos, BlockHitResult hitVec) {
             super(player, hand, pos, hitVec.getDirection());
@@ -215,4 +233,6 @@
     public static class RightClickItem extends PlayerInteractEvent
     {
+        public RightClickItem() {}
+
         public RightClickItem(Player player, InteractionHand hand)
         {
@@ -228,4 +248,6 @@
     public static class RightClickEmpty extends PlayerInteractEvent
     {
+        public RightClickEmpty() {}
+
         public RightClickEmpty(Player player, InteractionHand hand)
         {
@@ -252,4 +274,6 @@
         private Result useItem = DEFAULT;
 
+        public LeftClickBlock() {}
+
         public LeftClickBlock(Player player, BlockPos pos, Direction face)
         {
@@ -302,4 +326,6 @@
     public static class LeftClickEmpty extends PlayerInteractEvent
     {
+        public LeftClickEmpty() {}
+
         public LeftClickEmpty(Player player)
         {
