--- PlayerXpEvent.java
+++ PlayerXpEvent.java
@@ -20,4 +20,5 @@
 public class PlayerXpEvent extends PlayerEvent
 {
+    public PlayerXpEvent() {}
 
     public PlayerXpEvent(Player player)
@@ -36,4 +37,8 @@
         private final ExperienceOrb orb;
 
+        public PickupXp() {
+            orb = null;
+        }
+
         public PickupXp(Player player, ExperienceOrb orb)
         {
@@ -59,4 +64,6 @@
         private int amount;
 
+        public XpChange() {}
+
         public XpChange(Player player, int amount)
         {
@@ -87,4 +94,6 @@
         private int levels;
 
+        public LevelChange() {}
+
         public LevelChange(Player player, int levels)
         {
