--- PlayerDestroyItemEvent.java
+++ PlayerDestroyItemEvent.java
@@ -54,4 +54,10 @@
     @Nullable
     private final InteractionHand hand; // May be null if this player destroys the item by any use besides holding it.
+
+    public PlayerDestroyItemEvent() {
+        original = ItemStack.EMPTY;
+        hand = null;
+    }
+
     public PlayerDestroyItemEvent(Player player, @NotNull ItemStack original, @Nullable InteractionHand hand)
     {
