--- KeyMappingLookup.java
+++ KeyMappingLookup.java
@@ -8,4 +8,5 @@
 import com.mojang.blaze3d.platform.InputConstants;
 import net.minecraft.client.KeyMapping;
+import net.minecraftforge.client.extensions.IForgeKeyMapping;
 import org.jetbrains.annotations.Nullable;
 
@@ -51,5 +52,5 @@
             for (KeyMapping binding : bindings)
             {
-                if (binding.isActiveAndMatches(keyCode))
+                if (((IForgeKeyMapping) binding).isActiveAndMatches(keyCode))
                 {
                     return binding;
@@ -76,5 +77,5 @@
     public void put(InputConstants.Key keyCode, KeyMapping keyBinding)
     {
-        KeyModifier keyModifier = keyBinding.getKeyModifier();
+        KeyModifier keyModifier = ((IForgeKeyMapping) keyBinding).getKeyModifier();
         Map<InputConstants.Key, Collection<KeyMapping>> bindingsMap = map.get(keyModifier);
         Collection<KeyMapping> bindingsForKey = bindingsMap.get(keyCode);
@@ -89,6 +90,6 @@
     public void remove(KeyMapping keyBinding)
     {
-        KeyModifier keyModifier = keyBinding.getKeyModifier();
-        InputConstants.Key keyCode = keyBinding.getKey();
+        KeyModifier keyModifier = ((IForgeKeyMapping) keyBinding).getKeyModifier();
+        InputConstants.Key keyCode = ((IForgeKeyMapping) keyBinding).getKey();
         Map<InputConstants.Key, Collection<KeyMapping>> bindingsMap = map.get(keyModifier);
         Collection<KeyMapping> bindingsForKey = bindingsMap.get(keyCode);
