--- GrindstoneEvent.java
+++ GrindstoneEvent.java
@@ -18,4 +18,8 @@
     private int xp;
 
+    public GrindstoneEvent() {
+        top = bottom = null;
+    }
+
     protected GrindstoneEvent(ItemStack top, ItemStack bottom, int xp)
     {
@@ -84,4 +88,6 @@
         private ItemStack output;
 
+        public OnplaceItem() {}
+
         public OnplaceItem(ItemStack top, ItemStack bottom, int xp)
         {
@@ -125,4 +131,6 @@
         private ItemStack newBottom = ItemStack.EMPTY;
 
+        public OnTakeItem() {}
+
         public OnTakeItem(ItemStack top, ItemStack bottom, int xp)
         {
