--- ItemAttributeModifierEvent.java
+++ ItemAttributeModifierEvent.java
@@ -37,4 +37,10 @@
     private Multimap<Attribute, AttributeModifier> modifiableModifiers;
 
+    public ItemAttributeModifierEvent() {
+        stack = null;
+        slotType = null;
+        originalModifiers = null;
+    }
+
     public ItemAttributeModifierEvent(ItemStack stack, EquipmentSlot slotType, Multimap<Attribute, AttributeModifier> modifiers)
     {
