--- ItemStackedOnOtherEvent.java
+++ ItemStackedOnOtherEvent.java
@@ -44,4 +44,12 @@
     private final SlotAccess carriedSlotAccess;
 
+    public ItemStackedOnOtherEvent() {
+        carriedItem = stackedOnItem = null;
+        slot = null;
+        action = null;
+        player = null;
+        carriedSlotAccess = null;
+    }
+
     @ApiStatus.Internal
     public ItemStackedOnOtherEvent(ItemStack carriedItem, ItemStack stackedOnItem, Slot slot, ClickAction action, Player player, SlotAccess carriedSlotAccess)
