--- PlayLevelSoundEvent.java
+++ PlayLevelSoundEvent.java
@@ -48,4 +48,9 @@
     private float newPitch;
 
+    public PlayLevelSoundEvent() {
+        level = null;
+        originalPitch = originalVolume = 0F;
+    }
+
     public PlayLevelSoundEvent(@NotNull Level level, @NotNull SoundEvent sound, @NotNull SoundSource source, float volume, float pitch)
     {
@@ -166,4 +171,8 @@
         private final Entity entity;
 
+        public AtEntity() {
+            entity = null;
+        }
+
         public AtEntity(Entity entity, SoundEvent sound, SoundSource source, float volume, float pitch)
         {
@@ -196,4 +205,8 @@
         private final Vec3 position;
 
+        public AtPosition() {
+            position = null;
+        }
+
         public AtPosition(Level level, Vec3 position, SoundEvent sound, SoundSource source, float volume, float pitch)
         {
