--- RegisterCommandsEvent.java
+++ RegisterCommandsEvent.java
@@ -27,4 +27,10 @@
     private final Commands.CommandSelection environment;
     private final CommandBuildContext context;
+
+    public RegisterCommandsEvent() {
+        dispatcher = null;
+        environment = null;
+        context = null;
+    }
     
     public RegisterCommandsEvent(CommandDispatcher<CommandSourceStack> dispatcher, Commands.CommandSelection environment, CommandBuildContext context)
