--- TickEvent.java
+++ TickEvent.java
@@ -28,4 +28,10 @@
     public final LogicalSide side;
     public final Phase phase;
+
+    public TickEvent() {
+        type = null;
+        side = null;
+        phase = null;
+    }
     public TickEvent(Type type, LogicalSide side, Phase phase)
     {
@@ -39,4 +45,9 @@
         private final MinecraftServer server;
 
+        public ServerTickEvent() {
+            haveTime = null;
+            server = null;
+        }
+
         public ServerTickEvent(Phase phase, BooleanSupplier haveTime, MinecraftServer server)
         {
@@ -66,4 +77,5 @@
 
     public static class ClientTickEvent extends TickEvent {
+        public ClientTickEvent() {}
         public ClientTickEvent(Phase phase)
         {
@@ -76,4 +88,9 @@
         private final BooleanSupplier haveTime;
 
+        public LevelTickEvent() {
+            level = null;
+            haveTime = null;
+        }
+
         public LevelTickEvent(LogicalSide side, Phase phase, Level level, BooleanSupplier haveTime)
         {
@@ -98,4 +115,8 @@
         public final Player player;
 
+        public PlayerTickEvent() {
+            player = null;
+        }
+
         public PlayerTickEvent(Phase phase, Player player)
         {
@@ -107,4 +128,9 @@
     public static class RenderTickEvent extends TickEvent {
         public final float renderTickTime;
+
+        public RenderTickEvent() {
+            renderTickTime = 1F;
+        }
+
         public RenderTickEvent(Phase phase, float renderTickTime)
         {
