--- ForgeRegistryTagsProvider.java
+++ ForgeRegistryTagsProvider.java
@@ -13,6 +13,7 @@
 import net.minecraftforge.registries.IForgeRegistry;
 import org.jetbrains.annotations.Nullable;
+import xyz.bluspring.kilt.injections.data.tags.TagsProviderInjection;
 
-public abstract class ForgeRegistryTagsProvider<T> extends TagsProvider<T>
+public abstract class ForgeRegistryTagsProvider<T> extends TagsProvider<T> implements TagsProviderInjection
 {
     private static <T> Registry<T> wrapRegistry(IForgeRegistry<T> forgeRegistry)
@@ -27,5 +28,6 @@
     public ForgeRegistryTagsProvider(DataGenerator generator, IForgeRegistry<T> forgeRegistry, String modId, @Nullable ExistingFileHelper existingFileHelper)
     {
-        super(generator, wrapRegistry(forgeRegistry), modId, existingFileHelper);
+        super(generator, wrapRegistry(forgeRegistry));
+        this.kilt$addConstructorArgs("forge", existingFileHelper);
     }
 }
