--- ForgeRegistryTagsProvider.java
+++ ForgeRegistryTagsProvider.java
@@ -13,6 +13,7 @@
 import net.minecraftforge.registries.IForgeRegistry;
 import org.jetbrains.annotations.Nullable;
+import xyz.bluspring.kilt.remaps.data.tags.TagsProviderRemap;
 
-public abstract class ForgeRegistryTagsProvider<T> extends TagsProvider<T>
+public abstract class ForgeRegistryTagsProvider<T> extends TagsProviderRemap<T>
 {
     private static <T> Registry<T> wrapRegistry(IForgeRegistry<T> forgeRegistry)
