--- ForgeEntityTypeTagsProvider.java
+++ ForgeEntityTypeTagsProvider.java
@@ -6,15 +6,17 @@
 package net.minecraftforge.common.data;
 
+import net.minecraft.core.Registry;
 import net.minecraft.data.DataGenerator;
 import net.minecraft.data.tags.EntityTypeTagsProvider;
 import net.minecraft.world.entity.EntityType;
 import net.minecraftforge.common.Tags;
+import xyz.bluspring.kilt.remaps.data.tags.TagsProviderRemap;
 
-public class ForgeEntityTypeTagsProvider extends EntityTypeTagsProvider
+public class ForgeEntityTypeTagsProvider extends TagsProviderRemap<EntityType<?>>
 {
     
     public ForgeEntityTypeTagsProvider(DataGenerator generator, ExistingFileHelper existingFileHelper)
     {
-        super(generator, "forge", existingFileHelper);
+        super(generator, Registry.ENTITY_TYPE, "forge", existingFileHelper);
     }
 
