--- ForgeEntityTypeTagsProvider.java
+++ ForgeEntityTypeTagsProvider.java
@@ -6,15 +6,18 @@
 package net.minecraftforge.common.data;
 
+import net.minecraft.core.Registry;
 import net.minecraft.data.DataGenerator;
-import net.minecraft.data.tags.EntityTypeTagsProvider;
+import net.minecraft.data.tags.TagsProvider;
 import net.minecraft.world.entity.EntityType;
 import net.minecraftforge.common.Tags;
+import xyz.bluspring.kilt.injections.data.tags.TagsProviderInjection;
 
-public class ForgeEntityTypeTagsProvider extends EntityTypeTagsProvider
+public class ForgeEntityTypeTagsProvider extends TagsProvider<EntityType<?>> implements TagsProviderInjection
 {
     
     public ForgeEntityTypeTagsProvider(DataGenerator generator, ExistingFileHelper existingFileHelper)
     {
-        super(generator, "forge", existingFileHelper);
+        super(generator, Registry.ENTITY_TYPE);
+        this.kilt$addConstructorArgs("forge", existingFileHelper);
     }
 
