--- ForgeBiomeTagsProvider.java
+++ ForgeBiomeTagsProvider.java
@@ -6,6 +6,7 @@
 package net.minecraftforge.common.data;
 
+import net.minecraft.data.BuiltinRegistries;
 import net.minecraft.data.DataGenerator;
-import net.minecraft.data.tags.BiomeTagsProvider;
+import net.minecraft.data.tags.TagsProvider;
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.tags.BiomeTags;
@@ -14,11 +15,13 @@
 import net.minecraft.world.level.biome.Biomes;
 import net.minecraftforge.common.Tags;
+import xyz.bluspring.kilt.injections.data.tags.TagsProviderInjection;
 
-public final class ForgeBiomeTagsProvider extends BiomeTagsProvider
+public final class ForgeBiomeTagsProvider extends TagsProvider<Biome> implements TagsProviderInjection
 {
 
     public ForgeBiomeTagsProvider(DataGenerator arg, ExistingFileHelper existingFileHelper)
     {
-        super(arg, "forge", existingFileHelper);
+        super(arg, BuiltinRegistries.BIOME);
+        this.kilt$addConstructorArgs("forge", existingFileHelper);
     }
 
