--- QuadBakingVertexConsumer.java
+++ QuadBakingVertexConsumer.java
@@ -15,4 +15,5 @@
 import net.minecraft.core.Direction;
 import net.minecraftforge.client.textures.UnitTextureAtlasSprite;
+import xyz.bluspring.kilt.injections.client.render.model.BakedQuadInjection;
 
 import java.util.IdentityHashMap;
@@ -130,5 +131,5 @@
             return;
         // We have a full quad, pass it to the consumer and reset
-        quadConsumer.accept(new BakedQuad(quadData, tintIndex, direction, sprite, shade, hasAmbientOcclusion));
+        quadConsumer.accept(BakedQuadInjection.withAo(quadData, tintIndex, direction, sprite, shade, hasAmbientOcclusion));
         vertexIndex = 0;
         quadData = new int[QUAD_DATA_SIZE];
