--- IForgeTransformation.java
+++ IForgeTransformation.java
@@ -7,4 +7,5 @@
 
 import com.mojang.blaze3d.vertex.PoseStack;
+import io.github.fabricators_of_create.porting_lib.extensions.TransformationExtensions;
 import net.minecraft.core.Direction;
 
@@ -20,9 +21,9 @@
  */
 // TODO - 1.20: Transformation is not client-only, move this extension outside the client package
-public interface IForgeTransformation
+public interface IForgeTransformation extends TransformationExtensions
 {
     private Transformation self()
     {
-        return (Transformation) this;
+        return (Transformation) (Object) this;
     }
 
