--- NamespacedDefaultedWrapper.java
+++ NamespacedDefaultedWrapper.java
@@ -30,6 +30,7 @@
 import com.mojang.serialization.Lifecycle;
 import org.jetbrains.annotations.Nullable;
+import xyz.bluspring.kilt.injections.core.MappedRegistryInjection;
 
-class NamespacedDefaultedWrapper<T> extends DefaultedRegistry<T> implements ILockableRegistry, IHolderHelperHolder<T>
+class NamespacedDefaultedWrapper<T> extends DefaultedRegistry<T> implements ILockableRegistry, IHolderHelperHolder<T>, MappedRegistryInjection
 {
     private final ForgeRegistry<T> delegate;
@@ -53,5 +54,5 @@
 
         Validate.notNull(value);
-        markKnown();
+        ((MappedRegistryInjection) this).markKnown();
         this.elementsLifecycle = this.elementsLifecycle.add(lifecycle);
 
@@ -233,5 +234,9 @@
         public void onAdd(IForgeRegistryInternal<V> owner, RegistryManager stage, int id, ResourceKey<V> key, V value, V oldValue)
         {
-            owner.getSlaveMap(ID, NamespacedDefaultedWrapper.class).holders.onAdded(stage, id, key, value, oldValue);
+            var slaveMap = owner.getSlaveMap(ID, NamespacedDefaultedWrapper.class);
+            if (slaveMap == null)
+                return;
+
+            slaveMap.holders.onAdded(stage, id, key, value, oldValue);
         }
     }
