--- NamespacedHolderHelper.java
+++ NamespacedHolderHelper.java
@@ -42,4 +42,5 @@
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
+import xyz.bluspring.kilt.injections.HolderReferenceInjection;
 
 class NamespacedHolderHelper<T>
@@ -177,5 +178,5 @@
        if (this.holderLookup != null)
        {
-           List<Holder.Reference<T>> intrusive = this.holders.values().stream().filter(h -> h.getType() == Holder.Reference.Type.INTRUSIVE && !h.isBound()).toList();
+           List<Holder.Reference<T>> intrusive = this.holders.values().stream().filter(h -> ((HolderReferenceInjection) h).getType() == Holder.Reference.Type.INTRUSIVE && !h.isBound()).toList();
            if (!intrusive.isEmpty())
                throw new IllegalStateException("Some intrusive holders were not added to registry: " + intrusive);
@@ -265,5 +266,5 @@
     Holder<T> onAdded(RegistryManager stage, int id, ResourceKey<T> key, T newValue, T oldValue)
     {
-        if (stage != RegistryManager.ACTIVE && (this.holderLookup == null || !stage.isStaging()))  // Intrusive handlers need updating in staging.
+        if (stage != RegistryManager.ACTIVE && (this.holderLookup == null || !stage.getName().equals("STAGING")))  // Intrusive handlers need updating in staging.
             return null; // Only do holder shit on the active registries, not pending or snapshots.
 
