--- LanguageHook.java
+++ LanguageHook.java
@@ -9,4 +9,5 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonObject;
+import io.github.fabricators_of_create.porting_lib.mixin.common.accessor.MinecraftServerAccessor;
 import net.minecraft.locale.Language;
 import net.minecraft.server.packs.resources.Resource;
@@ -74,5 +75,5 @@
     private static void loadLanguage(String langName, MinecraftServer server) {
         String langFile = String.format(Locale.ROOT, "lang/%s.json", langName);
-        ResourceManager resourceManager = server.getServerResources().resourceManager();
+        ResourceManager resourceManager = ((MinecraftServerAccessor) server).port_lib$getServerResources().resourceManager();
         resourceManager.getNamespaces().forEach(namespace -> {
             try {
