--- TitleScreenModUpdateIndicator.java
+++ TitleScreenModUpdateIndicator.java
@@ -9,4 +9,5 @@
 import com.mojang.blaze3d.systems.RenderSystem;
 
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.screens.TitleScreen;
@@ -53,5 +54,5 @@
     public void render(PoseStack poseStack, int mouseX, int mouseY, float partialTick)
     {
-        if (showNotification == null || !showNotification.shouldDraw() || !FMLConfig.runVersionCheck())
+        if (showNotification == null || !showNotification.shouldDraw())
         {
             return;
@@ -71,5 +72,5 @@
     {
         TitleScreenModUpdateIndicator titleScreenModUpdateIndicator = new TitleScreenModUpdateIndicator(modButton);
-        titleScreenModUpdateIndicator.resize(guiMainMenu.getMinecraft(), guiMainMenu.width, guiMainMenu.height);
+        titleScreenModUpdateIndicator.resize(Minecraft.getInstance(), guiMainMenu.width, guiMainMenu.height);
         titleScreenModUpdateIndicator.init();
         return titleScreenModUpdateIndicator;
