--- FluidType.java
+++ FluidType.java
@@ -873,5 +873,5 @@
     {
         // Minecraft instance isn't available in datagen, so don't call initializeClient if in datagen
-        if (net.minecraftforge.fml.loading.FMLEnvironment.dist == net.minecraftforge.api.distmarker.Dist.CLIENT && !net.minecraftforge.fml.loading.FMLLoader.getLaunchHandler().isData())
+        if (net.minecraftforge.fml.loading.FMLEnvironment.dist == net.minecraftforge.api.distmarker.Dist.CLIENT)
         {
             initializeClient(properties ->
