--- ClientModLoader.java
+++ ClientModLoader.java
@@ -88,7 +88,10 @@
         LogicalSidedProvider.setClient(()->minecraft);
         LanguageHook.loadForgeAndMCLangs();
-        earlyLoaderGUI = new EarlyLoaderGUI(minecraft);
-        createRunnableWithCatch(()->ModLoader.get().gatherAndInitializeMods(ModWorkManager.syncExecutor(), ModWorkManager.parallelExecutor(), new SpacedRunnable(earlyLoaderGUI::renderTick))).run();
-        if (error == null) {
+        // Kilt: We don't need an early loader GUI.
+        //earlyLoaderGUI = new EarlyLoaderGUI(minecraft);
+        //createRunnableWithCatch(()->ModLoader.get().gatherAndInitializeMods(ModWorkManager.syncExecutor(), ModWorkManager.parallelExecutor(), new SpacedRunnable(earlyLoaderGUI::renderTick))).run();
+
+        // Kilt: Fabric already handles this for us.
+        if (false && error == null) {
             ResourcePackLoader.loadResourcePacks(defaultResourcePacks, ClientModLoader::buildPackFinder);
             ModLoader.get().postEvent(new AddPackFindersEvent(PackType.CLIENT_RESOURCES, defaultResourcePacks::addPackFinder));
@@ -127,5 +130,5 @@
         loadingComplete = true;
         // reload game settings on main thread
-        syncExecutor.execute(()->mc.options.load(true));
+        syncExecutor.execute(()->mc.options.load());
     }
 
