--- ForgeGui.java
+++ ForgeGui.java
@@ -188,5 +188,5 @@
     }
 
-    void renderHelmet(float partialTick, PoseStack poseStack)
+    public void renderHelmet(float partialTick, PoseStack poseStack)
     {
         ItemStack itemstack = this.minecraft.player.getInventory().getArmor(3);
@@ -246,5 +246,5 @@
 
     @Override
-    protected void renderPortalOverlay(float partialTick)
+    public void renderPortalOverlay(float partialTick)
     {
         float f1 = Mth.lerp(partialTick, this.minecraft.player.oPortalTime, this.minecraft.player.portalTime);
