--- EnchantmentLevelSetEvent.java
+++ EnchantmentLevelSetEvent.java
@@ -30,4 +30,13 @@
     private int enchantLevel;
 
+    public EnchantmentLevelSetEvent() {
+        level = null;
+        pos = null;
+        enchantRow = -1;
+        power = -1;
+        itemStack = ItemStack.EMPTY;
+        originalLevel = -1;
+    }
+
     public EnchantmentLevelSetEvent(Level level, BlockPos pos, int enchantRow, int power, @NotNull ItemStack itemStack, int enchantLevel)
     {
