--- EntityAttributeCreationEvent.java
+++ EntityAttributeCreationEvent.java
@@ -14,4 +14,5 @@
 import net.minecraftforge.eventbus.api.Event;
 import net.minecraftforge.fml.event.IModBusEvent;
+import xyz.bluspring.kilt.Kilt;
 
 /**
@@ -26,4 +27,8 @@
     private final Map<EntityType<? extends LivingEntity>, AttributeSupplier> map;
 
+    public EntityAttributeCreationEvent() {
+        map = null;
+    }
+
     public EntityAttributeCreationEvent(Map<EntityType<? extends LivingEntity>, AttributeSupplier> map)
     {
@@ -33,6 +38,10 @@
     public void put(EntityType<? extends LivingEntity> entity, AttributeSupplier map)
     {
-        if (DefaultAttributes.hasSupplier(entity))
-            throw new IllegalStateException("Duplicate DefaultAttributes entry: " + entity);
+        if (DefaultAttributes.hasSupplier(entity)) {
+            Kilt.Companion.getLogger().warn("Duplicate DefaultAttributes entry: " + entity);
+            Kilt.Companion.getLogger().warn("This will be ignored so Minecraft doesn't throw a fit.");
+
+            return;
+        }
         this.map.put(entity, map);
     }
