--- EntityEvent.java
+++ EntityEvent.java
@@ -13,4 +13,5 @@
 import net.minecraftforge.eventbus.api.Cancelable;
 import net.minecraftforge.eventbus.api.Event;
+import xyz.bluspring.kilt.mixin.EntityAccessor;
 
 /**
@@ -25,4 +26,8 @@
 public class EntityEvent extends Event
 {
+    public EntityEvent() {
+        entity = null;
+    }
+
     private final Entity entity;
 
@@ -49,4 +54,5 @@
     public static class EntityConstructing extends EntityEvent
     {
+        public EntityConstructing() {}
         public EntityConstructing(Entity entity)
         {
@@ -73,4 +79,8 @@
         private final long packedNewPos;
 
+        public EnteringSection() {
+            packedOldPos = packedNewPos = 0L;
+        }
+
         public EnteringSection(Entity entity, long packedOldPos, long packedNewPos)
         {
@@ -146,4 +156,10 @@
         private float newEyeHeight;
 
+        public Size() {
+            pose = null;
+            oldSize = null;
+            oldEyeHeight = 0F;
+        }
+
         public Size(Entity entity, Pose pose, EntityDimensions size, float defaultEyeHeight)
         {
@@ -178,5 +194,5 @@
             if (updateEyeHeight)
             {
-                this.newEyeHeight = this.getEntity().getEyeHeightAccess(this.getPose(), this.newSize);
+                this.newEyeHeight = ((EntityAccessor) this.getEntity()).callGetEyeHeight(this.getPose(), this.newSize);
             }
         }
