--- EntityAttributeModificationEvent.java
+++ EntityAttributeModificationEvent.java
@@ -15,4 +15,5 @@
 import net.minecraftforge.fml.event.IModBusEvent;
 import net.minecraftforge.registries.ForgeRegistries;
+import xyz.bluspring.kilt.injections.entity.AttributeSupplierBuilderInjection;
 
 import java.util.List;
@@ -32,4 +33,9 @@
     private final List<EntityType<? extends LivingEntity>> entityTypes;
 
+    public EntityAttributeModificationEvent() {
+        entityAttributes = null;
+        entityTypes = null;
+    }
+
     @SuppressWarnings("unchecked")
     public EntityAttributeModificationEvent(Map<EntityType<? extends LivingEntity>, AttributeSupplier.Builder> mapIn)
@@ -59,5 +65,5 @@
     {
         AttributeSupplier globalMap = DefaultAttributes.getSupplier(entityType);
-        return globalMap.hasAttribute(attribute) || (entityAttributes.get(entityType) != null && entityAttributes.get(entityType).hasAttribute(attribute));
+        return globalMap.hasAttribute(attribute) || (entityAttributes.get(entityType) != null && ((AttributeSupplierBuilderInjection) entityAttributes.get(entityType)).hasAttribute(attribute));
     }
 
