--- EntityTeleportEvent.java
+++ EntityTeleportEvent.java
@@ -36,4 +36,6 @@
     protected double targetZ;
 
+    public EntityTeleportEvent() {}
+
     public EntityTeleportEvent(Entity entity, double targetX, double targetY, double targetZ) {
         super(entity);
@@ -73,4 +75,6 @@
     public static class TeleportCommand extends EntityTeleportEvent
     {
+        public TeleportCommand() {}
+
         public TeleportCommand(Entity entity, double targetX, double targetY, double targetZ)
         {
@@ -97,4 +101,6 @@
     public static class SpreadPlayersCommand extends EntityTeleportEvent
     {
+        public SpreadPlayersCommand() {}
+
         public SpreadPlayersCommand(Entity entity, double targetX, double targetY, double targetZ)
         {
@@ -122,4 +128,8 @@
         private final LivingEntity entityLiving;
 
+        public EnderEntity() {
+            entityLiving = null;
+        }
+
         public EnderEntity(LivingEntity entity, double targetX, double targetY, double targetZ)
         {
@@ -157,4 +167,10 @@
         private final HitResult hitResult; // TODO - 1.20: make the hit result nonnull, remove the other constructor
 
+        public EnderPearl() {
+            player = null;
+            pearlEntity = null;
+            hitResult = null;
+        }
+
         @ApiStatus.Internal
         public EnderPearl(ServerPlayer entity, double targetX, double targetY, double targetZ, ThrownEnderpearl pearlEntity, float attackDamage, @Nullable HitResult hitResult)
@@ -223,4 +239,8 @@
         private final LivingEntity entityLiving;
 
+        public ChorusFruit() {
+            entityLiving = null;
+        }
+
         public ChorusFruit(LivingEntity entity, double targetX, double targetY, double targetZ)
         {
