--- SpawnPlacementRegisterEvent.java
+++ SpawnPlacementRegisterEvent.java
@@ -20,6 +20,6 @@
 import net.minecraftforge.eventbus.api.EventPriority;
 import net.minecraftforge.fml.event.IModBusEvent;
-import net.minecraftforge.registries.ForgeRegistries;
 
+import net.minecraftforge.registries.ForgeRegistries;
 import org.jetbrains.annotations.ApiStatus;
 
@@ -46,4 +46,8 @@
     private final Map<EntityType<?>, MergedSpawnPredicate<?>> map;
 
+    public SpawnPlacementRegisterEvent() {
+        map = null;
+    }
+
     @ApiStatus.Internal
     public SpawnPlacementRegisterEvent(Map<EntityType<?>, MergedSpawnPredicate<?>> map)
