--- ClientForgeMod.java
+++ ClientForgeMod.java
@@ -28,14 +28,14 @@
     {
         event.register("empty", EmptyModel.LOADER);
-        event.register("elements", ElementsModel.Loader.INSTANCE);
-        event.register("obj", ObjLoader.INSTANCE);
-        event.register("fluid_container", DynamicFluidContainerModel.Loader.INSTANCE);
-        event.register("composite", CompositeModel.Loader.INSTANCE);
-        event.register("item_layers", ItemLayerModel.Loader.INSTANCE);
+        //event.register("elements", ElementsModel.Loader.INSTANCE);
+        //event.register("obj", ObjLoader.INSTANCE);
+        //event.register("fluid_container", DynamicFluidContainerModel.Loader.INSTANCE);
+        //event.register("composite", CompositeModel.Loader.INSTANCE);
+        //event.register("item_layers", ItemLayerModel.Loader.INSTANCE);
         event.register("separate_transforms", SeparateTransformsModel.Loader.INSTANCE);
 
         // TODO: Deprecated. To be removed in 1.20
-        event.register("bucket", DynamicFluidContainerModel.Loader.INSTANCE_DEPRECATED);
-        event.register("item-layers", ItemLayerModel.Loader.INSTANCE_DEPRECATED);
+        //event.register("bucket", DynamicFluidContainerModel.Loader.INSTANCE_DEPRECATED);
+        //event.register("item-layers", ItemLayerModel.Loader.INSTANCE_DEPRECATED);
         event.register("separate-perspective", SeparateTransformsModel.Loader.INSTANCE_DEPRECATED);
     }
