--- RecipeBookManager.java
+++ RecipeBookManager.java
@@ -69,5 +69,5 @@
         var recipeCategoryLookups = new HashMap<RecipeType<?>, Function<Recipe<?>, RecipeBookCategories>>();
         var event = new RegisterRecipeBookCategoriesEvent(aggregateCategories, typeCategories, recipeCategoryLookups);
-        ModLoader.get().postEventWithWrapInModOrder(event, (mc, e) -> ModLoadingContext.get().setActiveContainer(mc), (mc, e) -> ModLoadingContext.get().setActiveContainer(null));
+        ModLoader.get().kiltPostEventWrappingMods(event);
         AGGREGATE_CATEGORIES.putAll(aggregateCategories);
         TYPE_CATEGORIES.putAll(typeCategories);
