--- ForgeSpawnEggItem.java
+++ ForgeSpawnEggItem.java
@@ -87,5 +87,5 @@
 
     @Mod.EventBusSubscriber(modid = "forge", bus = Mod.EventBusSubscriber.Bus.MOD)
-    private static class CommonHandler
+    public static class CommonHandler
     {
         @SubscribeEvent
@@ -106,5 +106,5 @@
 
     @Mod.EventBusSubscriber(value = Dist.CLIENT, modid = "forge", bus = Mod.EventBusSubscriber.Bus.MOD)
-    private static class ColorRegisterHandler
+    public static class ColorRegisterHandler
     {
         @SubscribeEvent(priority = EventPriority.HIGHEST)
